generators/entity-server/templates/dotnetcore/src/Project.Infrastructure/Data/ApplicationDatabaseContext.cs.ejs

```diff
@@ -126,16 +112,28 @@ namespace <%= namespace %>.Infrastructure.Data {
                 if(entity.name.toUpperCase() !== entityClass.toUpperCase()) {
                     entity.definition.relationships.forEach(relationship => {
                         if (relationship.relationshipType === 'many-to-many') {
+                            // executed when importing entities by jdl (jhipster import-jdl fileName.jdl)
                             if (relationship.ownerSide) {
                                 let relationshipNamePascalized = toPascalCase(relationship.relationshipName);
+                                let relationshipNamePascalizedPlural = pluralize(relationshipNamePascalized);
```

Comment:
The varible is not define here ? (https://github.com/jhipster/jhipster-dotnetcore/blob/main/generators/entity/index.js#L119) 
if not can you declare here once

Reply:
I have added the variable to generators/entity/index.js:
`relationship.relationshipNamePascalizedPlural = pluralize(relationship.relationshipNamePascalized);` 

BUT:
When running `jhipster import-jd fileName.jdl` when I try to access the variable at the ApplicationDatabaseContext.cs.ejs template, I can only access what is stored at the json file for that entity:
```
{
  "name": "Job",
  "fields": [
    {
      "fieldName": "jobTitle",
      "fieldType": "String"
    },
    ...
  ],
  "relationships": [
    {
      "relationshipType": "many-to-many",
      "otherEntityName": "pieceOfWork",
      "otherEntityRelationshipName": "job",
      "relationshipName": "chore",
      "otherEntityField": "title",
      "ownerSide": true
    },
    ...
```

That is why there is code duplication here... I don't know if it is a bug, if the variables are not generated (maybe because the generators/entity/index.js is called by the entity generator, not the import-jdl generator?)

When running the entity generator `jhipster entity job` and regenerating relationships the variables are accessible, but running `jhipster import-jdl fileName.jdl` the variables are not available

Reply:
import jdl use the same code than jhispter entity so and don't understand why your field is not accesible. Have you run the code in debug ?

Reply:
context.relationships loads the relationships only for the current entity being added using the entity generator, or the last entity processed by import-jdl. 
It is not useful for the use case of getting all the relationships of all entities, needed at ApplicationDatabaseContext.cs template file.

Now, context.entities has some data that can help with that:

context.entities stores all entities:
![image](https://user-images.githubusercontent.com/891084/99588070-d0e81200-29c8-11eb-93e0-b5f31b01ef39.png)

Each entity contains a definition, with relationship data (only the fields stored at .jhipster folder's json contents:
![image](https://user-images.githubusercontent.com/891084/99587919-9c745600-29c8-11eb-81cd-cfc68798f926.png)

Unfortunately it only has the "basic" info about relationships, same as stored at .jhipster/entityName.json not the in-memory data loaded.

That is why it was needed to "build" that data here.

Reply:
ah ok I understand this problem happens because the file is rewritten to each entity. 
A better way to do this is to use "needle" like here (https://github.com/jhipster/jhipster-dotnetcore/blob/main/generators/server/needle-api/needle-server-gateway.js) or here (https://github.com/jhipster/jhipster-dotnetcore/blob/main/generators/client/needle-api/needle-client-blazor.js) 
to write only a part of the file. 
There are two possibilities: we can take advantage of this issue to deal with this point or we transfer it to another issue.
It's like you want but if this point is treated in another issue I'm afraid it will never be treated

Reply:
I'm familiarizing with needles.

From the examples you provided, it seems to handle adding lines to an existing file.

Does it handle removing lines, like when a relationship is removed and we need removing the mapping from ApplicationDatabaseContext?

Reply:
I would say no, but I'm not sure.

Reply:
if is not possible, i think we can keep the current implementation

Reply:
I'm not sure too.

Maybe I can investigate it in another issue?

Reply:
what I propose to you is to stay like this for the moment. I have no way to delete entity/relation with needle